pkg/license/reporter_test.go

```diff
@@ -0,0 +1,221 @@
+									Name: essettings.EnvEsJavaOpts, Value: "-Xms8G -Xmx8G",
+								}},
+							},
+						},
+					},
+				},
+			}},
+		},
+	}
+	licensingInfo, err = NewResourceReporter(k8s.FakeClient(&es)).Get()
+	assert.NoError(t, err)
+	assert.Equal(t, "171.80GB", licensingInfo.TotalManagedMemory)
+	assert.Equal(t, "3", licensingInfo.EnterpriseResourceUnits)
+
+	kb := kbv1.Kibana{
+		Spec: kbv1.KibanaSpec{
+			Count: 100,
+		},
+	}
+	licensingInfo, err = NewResourceReporter(k8s.FakeClient(&kb)).Get()
+	assert.NoError(t, err)
+	assert.Equal(t, "107.37GB", licensingInfo.TotalManagedMemory)
+	assert.Equal(t, "2", licensingInfo.EnterpriseResourceUnits)
+
+	kb = kbv1.Kibana{
+		Spec: kbv1.KibanaSpec{
+			Count: 100,
+			PodTemplate: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: kbv1.KibanaContainerName,
+							Resources: corev1.ResourceRequirements{
+								Limits: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceMemory: resource.MustParse("2Gi"),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	licensingInfo, err = NewResourceReporter(k8s.FakeClient(&kb)).Get()
+	assert.NoError(t, err)
+	assert.Equal(t, "214.75GB", licensingInfo.TotalManagedMemory)
+	assert.Equal(t, "4", licensingInfo.EnterpriseResourceUnits)
+
+	kb = kbv1.Kibana{
+		Spec: kbv1.KibanaSpec{
+			Count: 100,
+			PodTemplate: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: kbv1.KibanaContainerName,
+							Env: []corev1.EnvVar{{
+								Name: kbconfig.EnvNodeOpts, Value: "--max-old-space-size=2048",
+							}},
+						},
+					},
+				},
+			},
+		},
+	}
+	licensingInfo, err = NewResourceReporter(k8s.FakeClient(&kb)).Get()
+	assert.NoError(t, err)
+	assert.Equal(t, "204.80GB", licensingInfo.TotalManagedMemory)
+	assert.Equal(t, "4", licensingInfo.EnterpriseResourceUnits)
+}
+
+func Test_Start(t *testing.T) {
+	es := esv1.Elasticsearch{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "es-test",
+		},
+		Spec: esv1.ElasticsearchSpec{NodeSets: []esv1.NodeSet{{Count: 20}}}}
+	kb := kbv1.Kibana{Spec: kbv1.KibanaSpec{Count: 2}}
+	apm := apmv1.ApmServer{Spec: apmv1.ApmServerSpec{Count: 2}}
+	k8sClient := k8s.FakeClient(&es, &kb, &apm)
+	operatorNs := "test-system"
+	refreshPeriod := 1 * time.Second
+
+	// start the resource reporter
+	go NewResourceReporter(k8sClient).Start(operatorNs, refreshPeriod)
+
+	// check that the licensing config map exists
+	assert.Eventually(t, func() bool {
+		var cm corev1.ConfigMap
+		err := k8sClient.Get(context.Background(), types.NamespacedName{
+			Namespace: operatorNs,
+			Name:      licensingCfgMapName,
+		}, &cm)
+		if err != nil {
+			return false
+		}
+		return cm.Data["timestamp"] != "" &&
+			cm.Data["eck_license_level"] == defaultOperatorLicenseLevel &&
+			cm.Data["enterprise_resource_units"] == "2" &&
+			cm.Data["total_managed_memory"] == "89.12GB"
+	}, refreshPeriod*2, refreshPeriod/2)
```

Comment:
`refreshPeriod*2 ` should be enough unless the system is under heavy load in CI. I would suggest we make it a bit higher than that.
We can also use our own `test.RetryUntilSuccess` which has its own defaults (100ms period & 5 sec timeout).